lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,203 @@
+                }
+            },
+
+            listenForKeypress: function() {
+
+                _dropdown.opts.page.on('keydown', function(e) {
+                    var keyCode = e.keyCode,
+                        focused = $(e.currentTarget.activeElement),
+                        items, menu;
+
+                    if (27 === keyCode) {
+                        // When the ESC key is pressed, close all menus
+                        _dropdown.closeDropdownMenus();
+                    }
+
+                    if (focused.is('.action')) {
+                        // Key handlers for when a menu item has focus
+                        menu = focused.closest('.dropdown-menu');
+                        _dropdown.handlerIsAction(keyCode, menu);
+
+                    } else if (focused.is('.has-dropdown')) {
+                        // Key handlers for when the button that opens the menu has focus
+                        _dropdown.handlerIsButton(keyCode, focused);
+
+                    } else if (focused.is('.dropdown-menu')) {
+                        // Key handlers for when the menu itself has focus, before an item within it receives focus
+                        menu = focused.closest('.dropdown-menu');
+                        _dropdown.handlerIsMenu(keyCode, menu);
+                    }
+                });
+            },
+
+            previousMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    index = items.index(focused),
+                    prev = index - 1;
+
+                if (index === 0) {
+                    items.last().focus();
+                } else {
+                    items.eq(prev).focus();
+                }
+            },
+
+            nextMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length -1,
+                    index = items.index(focused),
+                    next = index + 1;
+
+                if (index === items_count) {
+                    items.first().focus();
+                } else {
+                    items.eq(next).focus();
+                }
+            },
+
+            focusFirstItem: function(menu) {
+                menu.find('.dropdown-item:first .action').focus();
+            },
+
+            closeDropdownMenu: function() {
+                var open = _dropdown.opts.page.find(_dropdown.opts.menu).not(':focus');
+
+                if (open) {
+                    open.removeClass(_dropdown.opts.menu_active)
+                        .addClass(_dropdown.opts.menu_inactive);
+
+                    open.parent()
+                        .find(_dropdown.opts.button)
+                        .removeClass(_dropdown.opts.button_active)
+                        .attr('aria-expanded', 'false');
+                }
+            },
+
+            closeDropdownMenus: function() {
```

Comment:
It is not clear to me when one should use closeDropdownMenu vs. closeDropdownMenus-- please document this. Also, the code in the 2 methods is identical, except for the "not focus" part. Please refactor them to share code.